packages/db/src/proxy.ts

```diff
@@ -244,6 +268,47 @@ function deepEqual<T>(a: T, b: T): boolean {
     return true
   }
 
+  // Handle Temporal objects
+  // Check if both are Temporal objects of the same type
+  const aTag = (a as any)[Symbol.toStringTag]
+  const bTag = (b as any)[Symbol.toStringTag]
+
+  if (typeof aTag === `string` && typeof bTag === `string`) {
+    const temporalTypes = [
+      `Temporal.PlainDate`,
+      `Temporal.PlainTime`,
+      `Temporal.PlainDateTime`,
+      `Temporal.ZonedDateTime`,
+      `Temporal.Instant`,
+      `Temporal.PlainYearMonth`,
+      `Temporal.PlainMonthDay`,
+      `Temporal.Duration`,
+      `Temporal.TimeZone`,
+      `Temporal.Calendar`,
+    ]
+
+    const aIsTemporal = temporalTypes.includes(aTag)
+    const bIsTemporal = temporalTypes.includes(bTag)
+
+    if (aIsTemporal && bIsTemporal) {
+      // If they're different Temporal types, they're not equal
+      if (aTag !== bTag) return false
+
+      // Use Temporal's built-in equals method if available
+      if (typeof (a as any).equals === `function`) {
+        return (a as any).equals(b)
+      }
+
+      // For Duration, use toString comparison as it doesn't have equals
+      if (aTag === `Temporal.Duration`) {
```

Comment:
Since the default is to fallback to `toString` comparison (L308) there's no need for this if test.

Reply:
Actually, i wouldn't fallback to string comparison here but rather use the static `compare` method: `Temporal.Duration.compare(a, b) === 0`